### ISSUE TITLE
roadCLUS: error "Some nodes are not in the graph" when simulating roads using the 'mst' method

### ISSUE THREAD

**Issue by TylerMuhly** [bcgov/clus#348]
**The bug**
When running forestryCLUS with roadCLUS set to the 'mst' method it errors out with:

Jul26 13:34:57 blckng  total elpsd: 20 mins | 6 blockingCLUS UpdateBlocks 2
Jul26 13:34:57 blckng update the blocks table
Jul26 13:34:59 frstry  total elpsd: 20 mins | 6 forestryCLUS schedule 3
Jul26 13:34:59 frstry ...assigning zonal constraints
Jul26 13:35:08 frstry ...assigning adjacency constraint
Jul26 13:35:11 frstry central_group harvest Target: 7500000 
Jul26 13:35:11 frstry Using block priority
Jul26 13:35:14 rdCLUS  total elpsd: 20 mins | 6 roadCLUS buildRoads 7
Jul26 13:35:14 rdCLUS mstSolve
2022-07-26 13:35:14 ERROR::e: Some nodes are not in the graph
 2022-07-26 13:35:14 ERROR::e: cppRouting::get_distance_matrix(Graph = sim$g, from = landing.cell$landings, to = sim$roadSourceID, allcores = FALSE)
Error in cppRouting::get_distance_matrix(Graph = sim$g, from = landing.cell$landings,  : 
  Some nodes are not in the graph
  
Error is at roadCLUS.R Line #483, which is this fucntion:
    weights.closest.rd <- cppRouting::get_distance_matrix(Graph=sim$g, 
                                  from=landing.cell$landings, 
                                  to=sim$roadSourceID, 
                                  allcores=FALSE)

This is part of the mstSolve(sim) function at roadCLUS.R#105

To Reproduce
1. Run this scenario (https://github.com/bcgov/clus/blob/main/R/scenarios/central_group/bau_20220721.Rmd) with roadMethod = 'mst'. 
2. Input and output data is here: \\spatialfiles2.bcgov\archive\FOR\VIC\HTS\ANA\PROJECTS\CLUS\road_test

**Expected behavior**
Should simulate roads without this error.

**Comment by TylerMuhly** [comment]
I was able to look at the input objects (sim$g, landing.cell$landings and sim$roadSourceID), and they all have values (i.e., are not null). 

Error suggests that there are missing 'landings' in the graph. I tried some  joins to see where the issue might be; merge.test <- merge (g$data, landing.cell, by.x = "from", by.y = "landings", all.y = T) suggests that there are landings ("from)" that don;t have a "to" or "dist" in the graph.

So, I guess there are landings being missed in the graph? Not sure why some landings are being missed. Maybe a 'conflict' between cutblockseqprepCLUS and blockingCLUS? I am using both here to simulate historic roads. But the model runs blocking and then sequenceprep, so it should include both historic and potential landings. I can't pinpoint what the issue is.
